src/utils/time.cc

```diff
@@ -73,4 +75,12 @@ cr_timestamp_nsec_t GetUnixTimestampNsec() {
     return static_cast<cr_timestamp_nsec_t>(duration_cast<nanoseconds>(system_clock::now().time_since_epoch()).count());
 }
 
+int GetCurrentUtcDate() noexcept {
+    const time_t ts = time(nullptr);
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init,-warnings-as-errors)
+    struct tm    tm;
+    // Will always succeed since `ts` does not overflow
+    gmtime_r(&ts, &tm);
+    return (tm.tm_year + 1900) * 10000 + (tm.tm_mon + 1) * 100 + tm.tm_mday;
```

Comment:
Do you need an integer or it is ok to be string? generate an integer like this does not seem to be a good idea. You can't really do much calculations, and it will overflow if you put HHMMSS (even 64-bit unsigned int is not enough if you need microseconds).

Reply:
If you just want a count, IIRC there's a since-epoch func in chrono.
If you want a readable string, then use a fixed-size string.

Calendar and locale are 2 messy things that we should avoid getting into the details when possible.

Reply:
BTW说到这个我突然想起，历法钟从来就不是严格单调的哦，除了ntp的抖动，偶尔还有讨厌的润秒。

Reply:
If the main goal is about timezone, how about utc_clock?
https://en.cppreference.com/w/cpp/chrono/utc_clock

Reply:
> 目前gcc和clang都不支持utc_clock

那可以用gmtime那类的函数减一下，自己算个时区来，剩下的就可以丢给chrono了。

> 时间戳差个几秒钟几分钟会影响算法效果吗

有单调性假设的地方就是一个有和无的区别，是有本质不同的，这也就是为什么我们会到处检查乱序。

Reply:
我觉得其实时区不重要，rolling的目的只是隔一段时间切一个出来，那你只要能精确表示切割时间就足够了，没必要自己趟时区转换这个浑水？

Reply:
确实，对于切分其实没必要管时区。
我倒是有点在意我们收的数据都是UTC的吗？
除了交易所要留意，不同机子的localtime是不是有可能不一样啊？

Reply:
> 因为收到的每条数据都要比较日期（如果到了第二天就要新建目录/文件），所以我希望快点，字符串比较是没有整数快的

我之前还没看到这条。
如果按我之前忘记在哪说的，搞个独立的旁路router，专门负责开DB/加锁/改指针/关DB，其实就行了，切分逻辑怎么折腾对性能都几乎没有影响。

Reply:
Friendly reminder: do not mention non-core stuff in our core repo.

Also, you are still defining the core lib behaviors by a specific downstream use case.

另外，机器时区设置不同不意味着你需要处理时区转换。你把完整的时间信息放进分块就行了，这东西可以是包含时区信息很容易复原的时间戳，也可以是Unix timestamp。

Reply:
日期是通用，可是为什么是日期就很局限。你用十进制数表示年月日，那为什么不可以是年月日时分秒？为什么不可以到ms/us/us？答案是你表示不来，64位整数十进制位就那么多。所以这像是个为record rolling这个具体功能定制的半残函数。（更何况我觉得即便是rolling也不需要这个东西，拿字符串表示就行了，你需要的可能是 @xkszltl 说的旁路router）

Reply:
你指的“线上”做的是什么？我理解的是，线上最多能预处理出记录数据块的范围，线下要查询照样是要二分查找的。你指望线上直接切出严格的天级/小时级数据没那么靠谱，边界上总是会有一些不准确的，程序运行也不总是整天/整小时的，何况还有时区的问题。

Reply:
我是觉得如果你实在需要这种功能，那应该至少调整精度。但我怀疑我们根本不需要这个功能